### DISCUSSION TITLE
Remove switch list from w3init (or change how this is done)

### DISCUSSION THREAD

**Discussion by JessicaMeixner-NOAA** [NOAA-EMC/WW3#474] [General]
In the building of WW3 in ad3, we add the switches that were used to compile the model.  See [these lines] (https://github.com/NOAA-EMC/WW3/blob/develop/model/bin/ad3#L264-L280) which are copied below: 

```
# 2.e Add switches in w3initmd - - - - - - - - - - - - - - - - - - - - - -

    if [ "$name" = "w3initmd" ]
    then
      sw_1=''
      cp $path_w/w3initmd.$fext  $path_w/w3initmd.ft1
      for line in 1 2 3
        do
          sw_1=`echo $sw_str  | fold -w100 | sed -n "${line}p"`
          sw_old="PUT_SW$line"
          # echo $line  $sw_1 $sw_old
          sed -e "s/$sw_old/$sw_1/"  $path_w/w3initmd.ft1 > $path_w/w3initmd.ft2
          mv  $path_w/w3initmd.ft2 $path_w/w3initmd.ft1
        done
      mv $path_w/w3initmd.ft1 $path_w/w3initmd.$fext
      echo "'$path_w/$name.fts' '$name.$fext'"        >> w3adc.inp
    fi
```

Because of this build level code addition, it will make it harder if we want to move to other types of builds because of this custom code addition in the build itself.  We could get rid of adding the switches completely or add the code in the files themselves that will create the list of switches.     Or something else?

**Comment by JessicaMeixner-NOAA** [comment]
Any thoughts or preferences @aliabdolali @ukmo-ccbunney @mickaelaccensi @thesser1 ?

**Comment by JessicaMeixner-NOAA** [comment]
On further inspection the only place that the variable SWITCHES which is created via ad3 is output is in the netcdf files: 
See, ww3_ounf: 
```
ww3_ounf.ftn:151:      USE W3INITMD, ONLY: WWVER, SWITCHES
ww3_ounf.ftn:3647:        IRET=NF90_PUT_ATT(NCID,NF90_GLOBAL,'WAVEWATCH_III_switches',TRIM(SWITCHES))
```
 

Not having this manually inserted code in ad3 is going to help facilitate the transition to CPP flags, a few options: 
**Option 1**: Use some CPP flag/functionality to do this (I haven't found anything for this, but maybe @ukmo-ccbunney will) 
**Option 2**: Add code to ww3_ounf to create the set of switches maybe something like: 

```
!/ST1 SWITCHSRC='ST1 ' 
!/ST2 SWITCHSRC='ST2 ' 
!/ST3 SWITCHSRC='SR3 ' 

SWITCHPDLIB
!/PDLIB SWITCHPDLIB='PDLIB '


SWITCHES=SWITCHSRC//SWITCHPDLIB !with some better formatting and a more complete list of switches 

IRET=NF90_PUT_ATT(NCID,NF90_GLOBAL,'WAVEWATCH_III_switches',TRIM(SWITCHES))
```

**Option 3**: Remove the print line of switches from ww3_ounf and the SWITCHES variable, which will remove the need for ad3 to insert the values for SWITCHES manually. 

My preference is option 3 or if someone else knows a simple way for option 1 to work.  However, if it's important for any one to have the switches in the gridded output files, we always have a backup of option 2.

**Reply by thesser1** [reply]
@JessicaMeixner-NOAA  I do like having the switches listed in the netcdf files, but I could be supportive of removing them if it helps push this process forward.  If we remove them, I would like to revisit this issue after the CPP flags are fully functional to see if the model options could be added into the netcdf metadata at that point.

**Reply by JessicaMeixner-NOAA** [reply]
@thesser1 see: https://github.com/NOAA-EMC/WW3/pull/475#issuecomment-923817474 I think we'll be able to keep this.  If not I have a branch that removes it: https://github.com/JessicaMeixner-NOAA/WW3/tree/feature/noswitchlist
